### PR TITLE
HPCC4J-343 Nightly remote junit tests and .m2 caching for github actions

### PR DESCRIPTION
**Pull request by Michael-Gardner**
Signed-off-by: Michael Gardner <michael.gardner@lexisnexisrisk.com>

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [x] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).

## Checklist:
- [x] I have created a corresponding JIRA ticket for this submission
- [x] My code follows the code style of this project.
  - [ ] I have applied the Eclipse code-format template provided.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the HPCC Systems CONTRIBUTORS document (https://github.com/hpcc-systems/HPCC-Platform/wiki/Guide-for-contributors).
- [x] The change has been fully tested:
  - [ ] I have performed unit tests to cover my changes.
  - [ ] I have performed system test and covered possible regressions and side effects.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [x] Scalability
  - [x] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.

### CODE REVIEW

**Comment by Michael-Gardner** [comment]
@rpastrana Please take a look at this.  Nightly junit remote tests + added caching to the other active actions.

File: .github/workflows/nightlyTestSuite.yml [hunk 1]
```diff
@@ -25,15 +27,28 @@ jobs:
...
+    - name: Cache Maven packages
       uses: actions/cache@v2
       with:
         path: ~/.m2
         key: ${{ runner.os }}-m2-${{ hashFiles('**/pom.xml') }}
         restore-keys: ${{ runner.os }}-m2
   
     - name: Build with Maven
-      run: mvn -B --activate-profiles jenkins-on-demand -Dmaven.gpg.skip=true -Dmaven.test.failure.ignore=true install
+      run: mvn -B --activate-profiles jenkins-on-demand -Dmaven.gpg.skip=true -Dmaven.test.failure.ignore=true -Dmaven.javadoc.skip=true install
```
Comment (rpastrana) [inline .github/workflows/nightlyTestSuite.yml:54]: we prob don't want to ignore test failures

File: .github/workflows/nightlyTestSuite.yml [hunk 2]
```diff
@@ -25,15 +27,28 @@ jobs:
...
+        echo "PLATFORM_CONTAINER_NAME=$(docker ps --format \"{{.Names}}\")"  >> $GITHUB_ENV
+
+
+    - name: HPCCSystems Datafile Setup
+      run: |
+        echo "Platform container name = ${{ env.PLATFORM_CONTAINER_NAME }}"
+        wget -q https://raw.githubusercontent.com/Michael-Gardner/hpccsystems-java-project-testbed/main/generatedata.sh
+        chmod +x generatedata.sh
+        docker cp generatedata.sh ${{ env.PLATFORM_CONTAINER_NAME }}:/home/hpcc/generatedata.sh
+        docker exec -d ${{ env.PLATFORM_CONTAINER_NAME }} /home/hpcc/generatedata.sh
```
Comment (rpastrana) [inline .github/workflows/nightlyTestSuite.yml:43]: do we need to deploy certs?

File: .github/workflows/nightlyTestSuite.yml [hunk 3]
```diff
@@ -25,15 +27,28 @@ jobs:
     - name: Setup JDK 1.8
       uses: actions/setup-java@v1
       with:
-        java-version: 11.0.2
-    
+        java-version: 1.8
+
+    - name: Fetch hpccsystems-platform docker container name
```
Comment (rpastrana) [inline .github/workflows/nightlyTestSuite.yml:32]: as discussed off-line, we shouldn't target this docker container

**Review by rpastrana — APPROVED** [review]
@Michael-Gardner looks reasonable

**Pull request merged**